src/debug/di/CMakeLists.txt

```diff
@@ -85,5 +89,10 @@ endif(WIN32)
 
 add_precompiled_header(stdafx.h stdafx.cpp CORDBDI_SOURCES)
```

Comment:
It seems you can fix the issue just by moving the add_precompiled_header line e.g right after the `add_definitions(-MT)` line. I would prefer that instead of adding more `if` blocks at the end.

Reply:
Unfortunately it is not possible bcs sources are processed by `add_precompiled_header` and each file has compile flags added to it.  And these are exactly the flags which are raising warnigs during `masm` run.

https://github.com/dotnet/coreclr/blob/3743d9a7d5959383ae4c31b4d9273fb61e2fb389/functions.cmake#L138

Reply:
Right, that's why I've suggested moving the add_precompile_header before the point where the .asm file is added to the CORDBDI_SOURCES. But maybe I am missing something.

Reply:
Tried to correct location of calls but moving `add_precompiled_header` right after `add_definitions(-MT)` would require adding another call in `if (WIN32) else` code to support linux. Is it that what you want to achieve?

Reply:
Ah, I see what you mean. I have not noticed the add_precompiled_header is used for Unix too, since we don't use precompiled headers for Unix builds (the function is dummy). But we use this function everywhere for all platforms just relying on the check inside of it, so it is better to leave it here in the general code path.
So I guess that the only change I'd ask you to make is to remove the `set(CORDBDI_SOURCES ${CORDBDI_SOURCES})` which doesn't do anything together with the enclosing `if`. 
I was also considering making the `list(APPEND CORDBDI_SOURCES ${CORDBDI_SOURCES_ASM_FILE})` unconditional and setting the CORDBDI_SOURCES_ASM_FILE for ARM/ARM64 Windows to the generated .obj file path and to the .S file for Unix.